RespokeSDKTests/Functional/ConnectionDisconnectTests.m

```diff
@@ -0,0 +1,242 @@
+//
+//  ConnectionDisconnectTests.m
+//  RespokeSDK
+//
+//  Copyright 2015, Digium, Inc.
+//  All rights reserved.
+//
+//  This source code is licensed under The MIT License found in the
+//  LICENSE file in the root directory of this source tree.
+//
+//  For all details and documentation:  https://www.respoke.io
+//
+
+#import <Foundation/Foundation.h>
+
+#import <UIKit/UIKit.h>
+#import <XCTest/XCTest.h>
+#import "RespokeEndpoint+private.h"
+#import "Respoke+private.h"
+#import "RespokeClient+private.h"
+#import "RespokeTestCase.h"
+#import "RespokeGroup+private.h"
+#import "RespokeConnection+private.h"
+#import "RespokeGroupMessage.h"
+
+
+@interface ConnectionDisconnectTests : RespokeTestCase <RespokeClientDelegate, RespokeEndpointDelegate, RespokeGroupDelegate> {
+    RespokeGroup *firstClientGroup;
+    BOOL callbackDidSucceed;
+    BOOL doCopySelf;
+    RespokeEndpoint *recipientEndpoint;
+    RespokeEndpoint *senderEndpoint;
+}
+
+@end
+
+
+@implementation ConnectionDisconnectTests
+
+- (void)setUp
+{
+    [super setUp];
+    
+    // Put setup code here. This method is called before the invocation of each test method in the class.
+    callbackDidSucceed = NO;
+}
+
+
+- (void)tearDown
+{
+    // Put teardown code here. This method is called after the invocation of each test method in the class.
+    [super tearDown];
+}
+
+
+/**
+ *  This test should confirm that an error is thrown when sending an endpoint message when in the process of reconnecting
+ */
+- (void)testErrorWhenReconnectingEndpointMessage
+{
+    // Create a client to test with
+    NSString *recipientEndpointID = [RespokeTestCase generateTestEndpointID];
+    RespokeClient *recipientClient = [self createTestClientWithEndpointID:recipientEndpointID delegate:self];
+    
+    // Create a second client to test with
+    NSString *senderEndpointID = [RespokeTestCase generateTestEndpointID];
+    RespokeClient *senderClient = [self createTestClientWithEndpointID:senderEndpointID delegate:self];
+    
+    // Build references to each of the endpoints
+    recipientEndpoint = [senderClient getEndpointWithID:recipientEndpointID skipCreate:NO];
+    XCTAssertNotNil(recipientEndpoint, @"Should create endpoint instance");
+    recipientEndpoint.delegate = self;
+    
+    senderEndpoint = [recipientClient getEndpointWithID:senderEndpointID skipCreate:NO];
+    XCTAssertNotNil(senderEndpoint, @"Should create endpoint instance");
+    senderEndpoint.delegate = self;
+    
+    // diconnect and immediately reconnect
+    [senderClient disconnect];
```

Comment:
Opps.. I must have accidentally deleted the reconnect. It's in the other tests.  I'll re-add it.